### PR TITLE
Karim/rebase

### PR DESCRIPTION
**Pull request by KarimZS**

### CODE REVIEW

**Comment by KarimZS** [comment]
made a mistake!

**Comment by EmilStenstrom** [comment]
Nice to see your changes! If possible, I would love to get all the spelling fixes in, there are a lot of them! 😊 Mind adding them as a separate PR?

File: app.js [hunk 1]
```diff
@@ -128,13 +128,16 @@ function get_widgets(SETTINGS) {
...
+        {label: 'Last modified date', value: SETTINGS.algolia.index_name + '_lastmod_descending'}
       ]
     }),
     "clear": instantsearch.widgets.clearRefinements({
       container: '#clear-all',
       templates: {
-        resetLabel: 'Clear all'
+        resetLabel({ hasRefinements }, { html }) {
+          return html`<span>${hasRefinements ? 'Clear all' : ''}</span>`;
+        }
```
Comment (EmilStenstrom) [inline app.js:140]: Nice addition! Can you submit as PR too?! :)

File: app.js [hunk 2]
```diff
@@ -144,6 +147,8 @@ function get_widgets(SETTINGS) {
         attribute: 'categories',
         operator: 'and',
         showMore: true,
+        searchable: true,
+        searchableIsAlwaysActive: false
```
Comment (EmilStenstrom) [inline app.js:151]: This is also nice! So you can search on categories, would love to have this too!

**Comment by KarimZS** [comment]
I was actually trying to make great use of https://github.com/boglesby03/mybgg who made all these improvements! I don't want to take any of the credit! Maybe we can reach out and ask if we can integrate it in. Im testing it right now on my repo as 1 commit squashed from all their changes

File: app.js [hunk 3]
```diff
@@ -220,6 +260,46 @@ function get_widgets(SETTINGS) {
...
+    "refine_age": panel('Min age')(instantsearch.widgets.rangeSlider)(
+      {
+        container: '#facet-age',
+        attribute: 'min_age',
+        // max: 18,
+        // min: 0,
+        step: 1,
+        pips: false
+      }
+    ),
```
Comment (EmilStenstrom) [inline app.js:302]: What does this look like? I kind of like the idea of a slider!

**Comment by KarimZS** [comment]
I can verify/make it work and then open another pr once its confirmed if you'd like

File: app.js [hunk 4]
```diff
@@ -234,22 +314,35 @@ function get_widgets(SETTINGS) {
             num = match[2];
 
             type_callback = {
-              'best': function(num) { return '<strong>' + num + '</strong><span title="Best with">★</span>'; },
+              'best': function(num) { return '<span title="Best with"><strong>' + num + '</strong>★</span>'; },
               'recommended': function(num) { return num; },
-              'expansion': function(num) { return num + '<span title="With expansion">⊕</span>'; },
+              'expansion': function(num) { return '<span title="With expansion">' + num + '⊕</span>'; },
+              'supports': function(num) { return '<span title="Supported"><em>' + num + '~</em></span>'; },
+              'expansionsupport': function(num) { return '<span title="Supported With expansion"><em>' + num + '⊕~</em></span>'; }
```
Comment (EmilStenstrom) [inline app.js:321]: I like this too! We seem to have similar taste! :) Add as separate PR?

File: index.html [hunk 5]
```diff
@@ -5,9 +5,9 @@
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=0.7">
   <link rel="shortcut icon" type="image/x-icon" href="favicon.ico">
-  <script src="https://cdn.jsdelivr.net/npm/algoliasearch@4.1.0/dist/algoliasearch-lite.umd.js"></script>
-  <script src="https://cdn.jsdelivr.net/npm/instantsearch.js@4.3.1/dist/instantsearch.production.min.js"></script>
-  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.css@7.1.1/themes/reset-min.css">
+  <script src="https://cdn.jsdelivr.net/npm/algoliasearch@4/dist/algoliasearch-lite.umd.js"></script>
+  <script src="https://cdn.jsdelivr.net/npm/instantsearch.js@4/dist/instantsearch.production.min.js"></script>
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/instantsearch.css@7/themes/reset-min.css">
```
Comment (EmilStenstrom) [inline index.html:10]: Nice! I'd like to have auto updating packages too! Add this?

File: scripts/mybgg/bgg_client.py [hunk 6]
```diff
@@ -245,7 +357,8 @@ def log_item(_, item):
                                     ], required=False),
                                     hooks=xml.Hooks(after_parse=numplayers_to_result)
                                 )
-                            ]),
+                            ],
+                            required=False),
```
Comment (EmilStenstrom) [inline scripts/mybgg/bgg_client.py:361]: I guess this fixes some bug with some games not having suggested_numplayers? Would love this aswell.

File: scripts/mybgg/bgg_client.py [hunk 7]
```diff
@@ -275,8 +402,34 @@ def log_item(_, item):
                             attribute="value",
                             alias="rating"
                         ),
-                        xml.string("playingtime", attribute="value", alias="playing_time"),
-                        xml.string("minage", attribute="value", alias="min_age"),
+                        xml.string("playingtime", attribute="value", alias="playing_time", required=False),
```
Comment (EmilStenstrom) [inline scripts/mybgg/bgg_client.py:405]: What does required=False do here? If it fixes a bug I want it too! :)

File: scripts/mybgg/indexer.py [hunk 8]
```diff
@@ -198,34 +237,43 @@ def add_objects(self, collection):
 
                     game["color"] = f"{color_r}, {color_g}, {color_b}"
 
-            game["objectID"] = f"bgg{game['id']}"
+            game["objectID"] = f"bgg{game['collection_id']}"
```
Comment (EmilStenstrom) [inline scripts/mybgg/indexer.py:240]: What's the difference here?

File: scripts/mybgg/models.py [hunk 9]
```diff
@@ -44,6 +105,8 @@ def calc_playing_time(self, game_data):
             240: '3-4h',
         }
         for playing_time_max, playing_time in playing_time_mapping.items():
+            if not game_data["playing_time"]:
+                return 'Unknown'
```
Comment (EmilStenstrom) [inline scripts/mybgg/models.py:109]: Nice fix, want!

File: scripts/mybgg/models.py [hunk 10]
```diff
@@ -83,13 +146,118 @@ def calc_rating(self, game_data):
...
             0: "Light",
             1: "Light",
             2: "Light Medium",
             3: "Medium",
             4: "Medium Heavy",
             5: "Heavy",
         }
-        return weight_mapping[round(Decimal(game_data["weight"] or 0))]
+
+        return weight_mapping[round(Decimal(game_data["weight"] or -1))]
```
Comment (EmilStenstrom) [inline scripts/mybgg/models.py:166]: Nice fix!

**Comment by EmilStenstrom** [comment]
@KarimZS Looked through the code, there are lots of nice additions here! Added a comment to the ones that I think will benefit everyone. Would you mind submitting many small PR:s to get these in?

**Comment by KarimZS** [comment]
> @KarimZS Looked through the code, there are lots of nice additions here! Added a comment to the ones that I think will benefit everyone. Would you mind submitting many small PR:s to get these in?

@EmilStenstrom https://karimzs.github.io/mybgg/ is the page given these changes merged into my fork. Happy to make a pull request again from my master to here. Everything looks to be working but feel free to poke around and I can try and look/fix anything weird.

**Pull request closed without merging**